pkg/resources/api_integration.go

```diff
@@ -0,0 +1,340 @@
+		Computed: true,
+	},
+	// Computed. Info you get by issuing a 'DESCRIBE INTEGRATION <name>' command (AZURE_CONSENT_URL)
+	"azure_consent_url": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"api_allowed_prefixes": {
+		Type:        schema.TypeList,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Required:    true,
+		Description: "Explicitly limits external functions that use the integration to reference one or more HTTPS proxy service endpoints and resources within those proxies.",
+		MinItems:    1,
+	},
+	"api_blocked_prefixes": {
+		Type:        schema.TypeList,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Lists the endpoints and resources in the HTTPS proxy service that are not allowed to be called from Snowflake.",
+	},
+	"enabled": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "Specifies whether this API integration is enabled or disabled. If the API integration is disabled, any external function that relies on it will not work.",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "A description of the API integration.",
+	},
+	"created_on": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Date and time when the API integration was created.",
+	},
+}
+
+// APIIntegration returns a pointer to the resource representing an api integration
+func APIIntegration() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateAPIIntegration,
+		Read:   ReadAPIIntegration,
+		Update: UpdateAPIIntegration,
+		Delete: DeleteAPIIntegration,
+
+		Schema: apiIntegrationSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// CreateAPIIntegration implements schema.CreateFunc
+func CreateAPIIntegration(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	name := d.Get("name").(string)
+
+	stmt := snowflake.ApiIntegration(name).Create()
+
+	// Set required fields
+	stmt.SetBool(`ENABLED`, d.Get("enabled").(bool))
+
+	stmt.SetStringList("API_ALLOWED_PREFIXES", expandStringList(d.Get("api_allowed_prefixes").([]interface{})))
+
+	// Set optional fields
+	if v, ok := d.GetOk("comment"); ok {
+		stmt.SetString(`COMMENT`, v.(string))
+	}
+
+	if _, ok := d.GetOk("api_blocked_prefixes"); ok {
+		stmt.SetStringList("API_BLOCKED_PREFIXES", expandStringList(d.Get("api_blocked_prefixes").([]interface{})))
+	}
+
+	// Now, set the API provider
+	err := setAPIProviderSettings(d, stmt)
+	if err != nil {
+		return err
+	}
+
+	err = snowflake.Exec(db, stmt.Statement())
+	if err != nil {
+		return fmt.Errorf("error creating api integration: %w", err)
+	}
+
+	d.SetId(name)
+
+	return ReadAPIIntegration(d, meta)
+}
+
+// ReadAPIIntegration implements schema.ReadFunc
+func ReadAPIIntegration(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := d.Id()
```

Comment:
If you're going to set this id here, replace the `d.Id()`s below for consistency.

Reply:
Done